packages/kokkos-kernels/src/batched/KokkosBatched_Util.hpp

```diff
@@ -24,9 +24,7 @@
 
 #include "KokkosKernels_config.h"
 
-namespace KokkosBatched {
-
-  // TPL macros
+// TPL macros
 #if defined (KOKKOSKERNELS_ENABLE_TPL_MKL) 
 #define __KOKKOSBATCHED_INTEL_MKL__ 1
 #include "mkl_version.h"
```

Comment:
I'm not sure this fully resolves the issues if KokkosKernels is built with TPL_ENABLE_MKL=ON but KokkosKernels_ENABLE_TPL_MKL=OFF (which is the default).  In particular I don't see how it prevents the MKL  instantiations in KokkosKernels from being compiled.  Did you try a build in this configuration?

Reply:
The problem in the spgemm is that the code somehow sees mkl included in my code (with namespace KokkosBatched - this is my mistake). With MKL=ON or OFF, the mkl files should be included outside without a name space and the compiler does not confuse with the index object with kokkos batched namespace.

Reply:
Also, I was able to compile the code in trilinos. BTW, @ndellingwood don't we have a nightly test for TPLs ?

Reply:
> @ndellingwood don't we have a nightly test for TPLs

Not yet but we will this week, I put in the PR last week to update testing scripts that more easily allows TPLs to be enabled on the systems we run nightlies.